c10/util/SmallVector.h

```diff
@@ -1312,7 +1326,8 @@ class /* LLVM_GSL_OWNER */ SmallVector : public SmallVectorImpl<T>,
                   std::input_iterator_tag>,
           int> = 0>
   explicit SmallVector(Container&& c) : SmallVectorImpl<T>(N) {
-    this->append(c.begin(), c.end());
+    this->append(
+        std::forward<Container>(c).begin(), std::forward<Container>(c).end());
```

Comment:
Isn't this tehnically invalid since .end() or .begin() is a use-after-move?

Reply:
Usually they are different elements of the container (For example, two pointers). Therefore two different parts are moved..